test/generaloption.py

```diff
@@ -137,6 +148,23 @@ def test_help_long(self):
         self.assertTrue(topt.parser.help_to_file.getvalue().find("--level-longlevel") > -1,
                         "Long documentation expanded in long help")
 
+    def test_help_rst(self):
+        """Generate (long) help message"""
+        topt = TestOption1(go_args=['--help=rst'],
+                           go_nosystemexit=True,
+                           go_columns=100,
+                           help_to_string=True,
+                           prog='optiontest1',
+                           )
+        self.assertTrue(topt.parser.help_to_file.getvalue().find("``--level-longlevel``") > -1,
+                        "Long documentation in .rst format expanded in long help")
+
+    def test_help_wrong(self):
+        """Error when generating help message in wrong format."""
+        # FIXME: make this stricer (type + error message)
```

Comment:
better to test the allowed choices for existing methods, then to refine a test for `choices` itself